trading_calendars/exchange_calendar_xasx.py

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time
+
+from dateutil.relativedelta import MO
+from pandas import DateOffset
+from pandas.tseries.holiday import (
+    Holiday,
+    GoodFriday,
+    EasterMonday,
+    previous_friday,
+    weekend_to_monday,
+)
+from pytz import timezone
+
+from .common_holidays import (
+    new_years_day,
+    christmas,
+    weekend_christmas,
+    boxing_day,
+    weekend_boxing_day,
+)
+from .trading_calendar import HolidayCalendar, TradingCalendar
+
+NewYearsDay = new_years_day(observance=weekend_to_monday)
+
+AustraliaDay = Holiday(
+    'Australia Day',
+    month=1,
+    day=26,
+    observance=weekend_to_monday,
+)
+
+AnzacDay = Holiday(
+    'Anzac Day',
+    month=4,
+    day=25,
+    observance=weekend_to_monday,
+)
+
+QueensBirthday = Holiday(
+    "Queen's Birthday",
+    month=6,
+    day=1,
+    offset=[DateOffset(weekday=MO(2))],
+)
+
+LastTradingDayBeforeChristmas = Holiday(
+    'Last Trading Day Before Christmas',
+    month=12,
+    day=24,
+    start_date='2010',
+    observance=previous_friday,
+)
+Christmas = christmas()
+WeekendChristmas = weekend_christmas()
+BoxingDay = boxing_day()
+WeekendBoxingDay = weekend_boxing_day()
+
+LastTradingDayOfCalendarYear = Holiday(
+    'Last Trading Day Of Calendar Year',
+    month=12,
+    day=31,
+    start_date='2010',
+    observance=previous_friday,
+)
+
+
+class XASXExchangeCalendar(TradingCalendar):
+    """
+    Calendar for the Australian Securities Exchange.
+
+    Open Time: 10:00 AM, NSW (Sydney Time)
```

Comment:
Not a big deal, but is 'NSW' the actual name of the timezone? I had a hard time finding a reference to it.

Reply:
"NSW" is New South Wales, but looks like the time zone is Australian Eastern Time. I'll change.